src/main/java/ox/app/states/PreEndState.java

```diff
@@ -0,0 +1,107 @@
+package ox.app.states;
+
+import ox.app.game.ScoreBoard;
+import ox.app.game.Board;
+import ox.app.game.Player;
+import ox.app.utility.RoundBuffer;
+
+import java.util.function.Consumer;
+
+public class PreEndState implements GameState {
+
+    private final Board board;
+    private final RoundBuffer playerBuffer;
+    private final Player winningPlayer;
+    private final Consumer<String> output;
+    private final ScoreBoard scoreBoard;
+    private final int currentBoardSize;
+    private final int winningStroke;
+
+    private static final int POINTS_FOR_WIN = 3;
+    private static final int POINTS_FOR_DRAW = 1;
+    private static final int POINTS_FOR_LOST = 0;
+    private static final int NUMBER_OF_ROUNDS = 3;
+
+    private final String winnerMessage;
+    private static final String DRAW_MESSAGE = "It's a draw!";
+
+    private int roundCounter;
+
+    public PreEndState(RoundBuffer playerBuffer, Consumer<String> output, Board board, ScoreBoard scoreBoard, int roundCounter, int currentBoardSize,int winningStroke) {
```

Comment:
This is really BIG constructor.
You should think if you really need all of this stuff here.
If you do, you should try to create some helper classes

Reply:
For now i want to leave it that way. It will be last step for refactoring.

Reply:
Ok